src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -256,13 +255,17 @@ Copyright (c) .NET Foundation. All rights reserved.
   </Target>
 
   <Target Name="_CheckMicrosoftNetSdkCompilersToolsetPackageExists"
-          Condition="'$(_NeedToDownloadMicrosoftNetSdkCompilersToolsetPackage)' == 'true' and '$(DesignTimeBuild)' != 'true' and '$(_SkipCheckMicrosoftNetSdkCompilersToolsetPackageExists)' != 'true'"
+          Condition="'$(_NeedToDownloadMicrosoftNetSdkCompilersToolsetPackage)' == 'true' and '$(DesignTimeBuild)' != 'true'"
           BeforeTargets="CoreCompile">
     <!-- If users did not run restore or it failed to download the Microsoft.Net.Sdk.Compilers.Toolset package
          (but they proceeded to build, e.g., in Visual Studio), display an error with suggestions how to fix the problem. -->
     <NETSdkError ResourceName="MicrosoftNetSdkCompilersToolsetNotFound"
-        Condition="!Exists('$(RoslynTargetsPath)')"
+        Condition="'$(_MicrosoftNetSdkCompilersToolsetPackageRootEmpty)' != 'true' and !Exists('$(RoslynTargetsPath)')"
         FormatArguments="$(NETCoreSdkVersion)" />
+
+    <!-- Warn if $(NuGetPackageRoot) is empty. See https://github.com/dotnet/sdk/issues/43016. -->
+    <NETSdkWarning ResourceName="MicrosoftNetSdkCompilersToolsetRootEmpty"
+        Condition="'$(_MicrosoftNetSdkCompilersToolsetPackageRootEmpty)' == 'true'" />
```

Comment:
The previous error task has a `!Exists('$(RoslynTargetsPath)')` clause.  Should that be mirrored here, or not?

Reply:
I don't think that's necessary, even if `$(RoslynTargetsPath)` exists and `$(NuGetPackageRoot)` is empty, that feels like a situation which warrants a warning.

Reply:
This warning is guarded by `$(_NeedsToDownloadMicrosoftNetSdkCompilersToolsetPackage)` which means that we are using NuGet here. If we're using NuGet and the `$(NuGetPackageRoot)` is empty then that seems like a very good case for a warning. 

One consideration I made here is that `-p:RolynTargetsPath=` has been a valid customer scenario for 7+ years now. That still works without warning here even if there is an empty `$(NuGetPackageRoot)` because it doesn't end up down this code path.